src/chord.js

```diff
@@ -1,368 +1,108 @@
-  }
+    this.attrGetter = new AttrGetter(this.chordComponentRef);
 
-  getReflectAttr() {
-    return this.chordComponentRef.getAttribute("reflect");
-  }
+    if (this.attrGetter.chordArray.length) {
+      this.initAttributes();
 
-  resolveReflectAttr(attrStr) {
-    const reflectModel = { horizontal: false, vertical: false };
-    const reflectData =
-      attrStr && attrStr.length ? attrStr.toLowerCase().split(/\s+|[,.]+/) : "";
-    if (reflectData) {
-      reflectData.map(el => {
-        if (el == "horizontal" || el == "x") {
-          reflectModel.horizontal = true;
-        }
-        if (el == "vertical" || el == "y") {
-          reflectModel.vertical = true;
-        }
+      this.shadowRoot.innerHTML = this.resolveFinalTemplate();
+      this.customAttributes.chords.map(chord => {
+        const customAttributes = {
+          ...this.customAttributes,
+          reflection: this.canvasSettings.reflect
+        };
+        new CanvasDrawTool(
+          this.initCanvas(chord),
+          chord,
+          customAttributes
+        ).draw();
       });
+    } else {
+      this.shadowRoot.innerHTML =
+        this.templatesHolder.unknownChordTemplate +
+        stylesHolder.getUnknownChordTemplateStyleTag();
     }
-    this.canvasSettings.reflect = reflectModel;
   }
 
   initAttributes() {
-    this.resolveSize(this.getSizeAttr());
-
-    this.customAttributes.currentChord = this.getCurrentChord(
-      this.getChordNameAttr()
-    );
-    this.resolveTheme(this.getThemeAttr());
-
-    this.resolveReflectAttr(this.getReflectAttr());
+    this.initChords();
+    this.initSize();
+    this.initTheme();
+    this.initReflectAttr();
   }
 
-  resolveTheme(themeStr) {
-    this.customAttributes.theme =
-      themeStr && themeStr.toLowerCase() == "dark" ? "dark" : "light";
+  initSize() {
+    this.customAttributes.size = this.attrGetter.size;
   }
 
-  resolveSize(size) {
-    const sizeParsed = parseFloat("" + size);
-    if (isNaN(sizeParsed)) {
-      this.customAttributes.size = sizeList[size]
-        ? sizeList[size]
-        : sizeList.medium;
-    } else this.customAttributes.size = sizeParsed;
-    this.canvasSettings.canvasHeight =
-      this.canvasSettings.canvasHeight * this.customAttributes.size;
-    this.canvasSettings.canvasWidth =
-      this.canvasSettings.canvasWidth * this.customAttributes.size;
+  initChords() {
+    this.customAttributes.chords = this.attrGetter.chordArray;
   }
 
-  getCurrentChord(chordName) {
-    return chordList.find(
-      el => el.name.toUpperCase() == chordName.toUpperCase()
-    );
+  initReflectAttr() {
+    this.canvasSettings.reflect = this.attrGetter.reflect;
   }
 
-  initCanvas() {
-    this.canvasSettings.canvas = this.shadowRoot.querySelector("canvas");
-    this.canvasSettings.ctx = this.canvasSettings.canvas
-      ? this.canvasSettings.canvas.getContext("2d")
-      : null;
+  initTheme() {
+    this.customAttributes.theme = this.attrGetter.theme;
```

Comment:
shouldn't we handle here case when attrGetter.theme is null and aa a default value?